data/crac/crac-api/src/main/java/com/farao_community/farao/data/crac_api/usage_rule/OnStateToRemedialActionAdder.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.data.crac_api.usage_rule;
+
+import com.farao_community.farao.data.crac_api.RemedialAction;
+import com.farao_community.farao.data.crac_api.State;
+
+/**
+ * Adds a OnState usage rule to a RemedialAction
+ * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
+ */
+public interface OnStateToRemedialActionAdder<T extends RemedialAction<T>> {
```

Comment:
The naming is a bit confusing because a RemedialActionAdder is something that exists so my brain first read it as OnStateTo(RemedialActionAdder) rather than OnState(ToRemedialAction)Adder. 
Maybe OnStateAdderToRemedialAction is clearer

Reply:
yes true